izpack-maven-plugin/src/main/java/org/izpack/mojo/IzPackNewMojo.java

```diff
@@ -219,44 +273,37 @@ private File getJarFile()
 
     private void initMavenProperties(PropertyManager propertyManager)
     {
-    	//TODO - project is annotated as @required, so the check project!=null should be useless!?!
-        if (project != null)
+        Properties properties = project.getProperties();
+        Properties userProps  = session.getUserProperties();
+        for (String propertyName : properties.stringPropertyNames())
         {
-            Properties properties = project.getProperties();
-            Properties userProps  = session.getUserProperties();
-            String[] exclusionList = null;
-            if (excludeProperties != null) {
-                exclusionList = excludeProperties.split(",");
+            if (containsExcludedProperty(propertyName))
+            {
+                continue;
             }
-            for (String propertyName : properties.stringPropertyNames())
+            String value;
+            // TODO: should all user properties be provided as property?
+            // Intentionally user properties are searched for properties defined in pom.xml only
+            // see https://izpack.atlassian.net/browse/IZPACK-1402 for discussion
+            if (userProps.containsKey(propertyName))
             {
-                String value;
-                // TODO: should all user properties be provided as property?
-                // Intentionally user properties are searched for properties defined in pom.xml only
-                // see https://izpack.atlassian.net/browse/IZPACK-1402 for discussion
-                if (userProps.containsKey(propertyName))
-                {
-                    value = userProps.getProperty(propertyName);
-                } else {
-                    value = properties.getProperty(propertyName);
-                }
-                if (!containsExcludedProperty(propertyName, exclusionList))
-                {
-                    String existingValue = propertyManager.getProperty(propertyName);
-                    if (existingValue != null && existingValue.equals(value))
-                    {
-                        getLog().debug("Maven property exists: " + propertyName + "=" + value);
-                    }
-                    else if (propertyManager.addProperty(propertyName, value))
-                    {
-                        getLog().debug("Maven property added: " + propertyName + "=" + value);
-                    }
-                    else
-                    {
-                        getLog().warn("Property " + propertyName + "=" + existingValue +
-                                " could not be overridden with maven property " + propertyName + "=" + value);
-                    }
-                }
+                value = userProps.getProperty(propertyName);
+            } else {
+                value = properties.getProperty(propertyName);
+            }
```

Comment:
We could avoid the duplicate lookup by o-priori getting the property value and doing a `null` as check of existence:

```
String value = userProps.getProperty(propertyName);
// TODO: should all user properties be provided as property?
// Intentionally user properties are searched for properties defined in pom.xml only
// see https://izpack.atlassian.net/browse/IZPACK-1402 for discussion
if (value == null)
{
        value = properties.getProperty(propertyName);
}
```

Reply:
Done